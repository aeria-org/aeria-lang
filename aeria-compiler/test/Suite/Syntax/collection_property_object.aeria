collection CollectionPropertyObjectTest {
  properties {
    prop1 {
      properties {
        prop2 str
        prop3 num
        prop4 {
          properties {
            prop5 str
            prop6 str
          }
        }
      }
    }
  }
}
