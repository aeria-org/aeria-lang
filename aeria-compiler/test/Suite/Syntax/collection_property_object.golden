{
  "kind": "Program",
  "collections": [
    {
      "properties": [
        {
          "type": {
            "required": [],
            "properties": [
              {
                "type": {
                  "kind": "PString"
                },
                "name": {
                  "value": "prop2",
                  "kind": "PropertyName"
                },
                "kind": "Property",
                "attributes": []
              },
              {
                "type": {
                  "kind": "PNum"
                },
                "name": {
                  "value": "prop3",
                  "kind": "PropertyName"
                },
                "kind": "Property",
                "attributes": []
              },
              {
                "type": {
                  "required": [],
                  "properties": [
                    {
                      "type": {
                        "kind": "PString"
                      },
                      "name": {
                        "value": "prop5",
                        "kind": "PropertyName"
                      },
                      "kind": "Property",
                      "attributes": []
                    },
                    {
                      "type": {
                        "kind": "PString"
                      },
                      "name": {
                        "value": "prop6",
                        "kind": "PropertyName"
                      },
                      "kind": "Property",
                      "attributes": []
                    }
                  ],
                  "kind": "PObject"
                },
                "name": {
                  "value": "prop4",
                  "kind": "PropertyName"
                },
                "kind": "Property",
                "attributes": []
              }
            ],
            "kind": "PObject"
          },
          "name": {
            "value": "prop1",
            "kind": "PropertyName"
          },
          "kind": "Property",
          "attributes": []
        }
      ],
      "name": {
        "value": "CollectionPropertyObjectTest",
        "kind": "CollectionName"
      },
      "kind": "Collection"
    }
  ]
}
